src/services/symbolDisplay.ts

```diff
@@ -638,26 +648,49 @@ namespace ts.SymbolDisplay {
             displayParts.push(spacePart());
         }
 
-        function addFullSymbolName(symbolToDisplay: Symbol, enclosingDeclaration?: Node) {
+        function addFullSymbolName(symbolToDisplay: Symbol, enclosingDeclaration?: Node, indexInfos?: IndexInfo[]) {
             if (alias && symbolToDisplay === symbol) {
                 symbolToDisplay = alias;
             }
             const fullSymbolDisplayParts = symbolToDisplayParts(typeChecker, symbolToDisplay, enclosingDeclaration || sourceFile, /*meaning*/ undefined,
                 SymbolFormatFlags.WriteTypeParametersOrArguments | SymbolFormatFlags.UseOnlyExternalAliasing | SymbolFormatFlags.AllowAnyNodeKind);
+            if (symbolToDisplay.flags & SymbolFlags.Signature) {
+                if (indexInfos) {
+                    let index = 1;
+                    fullSymbolDisplayParts[index++] = punctuationPart(SyntaxKind.OpenBracketToken);

```

Comment:
Since fullSymbolDisplayParts is something like
```
[{text: 'TestInterface', kind: 'interfaceName'}, 
{text: '.', kind: 'punctuation'}, 
{text: '__index', kind: 'text'}] 
```
I wanted to keep fullSymbolDisplayParts [0] but not the rest, so I was keeping track of index, to do that. 
I know realize I can probably do fullSymbolDisplayParts = symbolToDisplayParts(typeChecker, symbolToDisplay.parent) instead. Would that be ok?

Reply:
Yeah, I think that sounds better. Alternatively, you could remove the parts you don’t want to keep before pushing, but that sounds a little sketchy. Better to avoid generating the pieces you plan to replace.